<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
</body>
<script>
    {
    	let arr = [1,2];
	    async function test1(){
		    const names = ['1', '2']
		    for (const name of names) {
			    const column = await leadsService(name)
			    console.log(column)
		    }
		    console.log(arr)
	    }
	    //test1();
	    function leadsService(){
		    return 1;
	    }
    }
    {
        function f(val) {
            // setTimeout(()=>{
            //     console.log(val);
            // }, 0 , val);
            console.log(val);
        }
        async function test(arr) {
            for(let i of arr){
                await f(i);
            }
            console.log(100);
        }
        test([1,2,3]);
    }
    console.log("********************************")
    {
        function f(val) {
            setTimeout(()=>{
                console.log(val);
            }, 0 , val);
        }
        async function test(arr) {
            for(let i of arr){
                await f(i);
            }
            console.log(100);
        }
      // test([1,2,3]);
    }
    {
        let promiseDemo = new Promise((resolve, reject) => {
            setTimeout(() => {
                let random = Math.random()
                if (random >= 0.5) {
                    resolve('success')
                } else {
                    reject('failed')
                }
            }, 1000)
        })

        async function test() {
            let result = await promiseDemo
            return result  //这里的result是promiseDemo成功状态的值，如果失败了，代码就直接跳到下面的catch了
        }

        // test().then(response => {
        //     console.log(response)
        // }).catch(error => {
        //     console.log(error)
        // })
    }

    {
        function fn(name) {
            return new Promise((resolve, reject) => {
                setTimeout(() => {
                    resolve(`${name}成功`)
                }, 1000)
            })
        }

        async function test() {
            return await Promise.all([fn('小红'), fn('小明'), fn('小华')])
        }

        // test().then(result => {
        //     console.log(result)
        // }).catch(result => {
        //     console.log(result)
        // })
    }

    {
        // console.log(1)
        // let promiseDemo = new Promise((resolve, reject) => {
        //     console.log(2)
        //     setTimeout(() => {
        //         let random = Math.random()
        //         if (random >= 0.2) {
        //             resolve('success')
        //             console.log(3)
        //         } else {
        //             reject('failed')
        //             console.log(3)
        //         }
        //     }, 1000)
        // })
        //
        // async function test() {
        //     console.log(4)
        //     let result = await promiseDemo
        //     return result
        // }
        //
        // test().then(result => {
        //     console.log(5)
        // }).catch((result) => {
        //     console.log(5)
        // })
        //
        // console.log(6)
    }

</script>
</html>